server/server.js

```diff
@@ -22,6 +21,19 @@ app.get('/getData/:itemId', (request, response) => {
     },request.params.itemId)
 })
 
+app.get('/search/:searchString', (request, response) => {
+    db.searchProducts((err, results) => {
+        if (err) {
+            console.log('GETPRODUCTDATAERROR: ', err);
+        }
+        else {
+            response.send(results);
+            console.log('sending data from server');
+        }
+        response.end();
```

Comment:
do you want this response.end to still be in your search products query, or could you put it outside?